packages/lockfile-lint-api/src/ParseLockfile.js

```diff
@@ -21,6 +21,9 @@ const {
  * @return boolean
  */
 function checkSampleContent (lockfile, isYarnBerry) {
+  if (Object.entries(lockfile).length < (isYarnBerry ? 2 : 1)) {
```

Comment:
what's actually happening here? are the object entries differ between the two versions? Wondering if there's another way of finding this out than that implementation detail.

Reply:
Yes, the entries differ between the two versions. I got that idea of using the length from the next line down: https://github.com/lirantal/lockfile-lint/blob/969ed058ed0037821ca5d681822e561dbe6d823c/packages/lockfile-lint-api/src/ParseLockfile.js#L24

Reply:
> I don't know if this actually fixes the issue either. I sounds like instead of the update here we should be updating `function yarnParseAndVerify (lockfileBuffer)` to assert whether the `lockfile` returned has anything in the object or not.

That could work... but "has anything in the object or not" varies between yarn berry and non-berry. And that conditional behavior seems to be in `checkSampleContent` (there's already that length conditional behavior on the `length`), which is why I did the checking there. And, the stack trace of the error that this PR seeks to address originates in `checkSampleContent` further indicating that it needs more input validation, leading me to believe that this fix belongs there for that reason as well.

> Also, `checkSampleContent()` is only used for Yarn...?

This bug only exists for yarn. I updated this PR to have 2 commits now to make that clearer:
* The first commit just tests npm behavior, to make sure it works with an empty lock file, which it does already.
* The second commits adds a test for empty yarn lock files, and also makes the changed needed so that test passes.